### PR TITLE
[enhance](partition_id) check partition id before store meta

### PR DESCRIPTION
**Pull request by dataroaring**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

File: be/src/olap/data_dir.cpp [hunk 1]
```diff
@@ -539,15 +539,23 @@ Status DataDir::load() {
...
                           << " for txn: " << rowset_meta->txn_id();
+
+            } else if (commit_txn_status.is<ErrorCode::INTERNAL_ERROR>()) {
+                LOG(WARNING) << "failed to add committed rowset: " << rowset_meta->rowset_id()
+                             << " to tablet: " << rowset_meta->tablet_id()
+                             << " for txn: " << rowset_meta->txn_id()
+                             << " error: " << commit_txn_status;
+                return commit_txn_status;
+            } else {
+                LOG(WARNING) << "failed to add committed rowset: " << rowset_meta->rowset_id()
```
Comment (xiaokang) [inline be/src/olap/data_dir.cpp:555]: just a warn log, no further error handling action?

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Review by deardeng — APPROVED** [review]
LTGM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**